packages/xo-web/src/xo-app/menu/index.js

```diff
@@ -25,12 +26,62 @@ import {
   getXoaState,
   isAdmin,
 } from 'selectors'
-import { every, identity, isEmpty, map } from 'lodash'
+import { every, forEach, identity, isEmpty, isEqual, map } from 'lodash'
 
 import styles from './index.css'
 
 const returnTrue = () => true
 
+@connectStore(() => ({
+  hosts: createGetObjectsOfType('host'),
+}))
+class MissingPatchWarning extends Component {
+  state: { haveMissingPatches: false }
+
+  componentWillMount() {
+    this._subscribeMissingPatches()
+  }
+
+  componentWillUnmount() {
+    this.unsubscribeMissingPatches()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!isEqual(prevProps.hosts, this.props.hosts)) {
+      this.setState({ haveMissingPatches: false })
```

Comment:
So if the hosts don't change, it will never be set to `false`?

Reply:
Yes. 
Now I compute the number of the missing patches to display the warning.